### PR TITLE
sonic-mgmt: fix t0-isolated-d128u128s2 topo

### PR DESCRIPTION
**Pull request by ccroy-arista**
### Description of PR
This PR contains the following changes, in order to achieve a functional t0-isolated-d128u128s2 topo:
- The field 'bp_interfaces' in the t0-isolated-d128u128s2 yml file is corrected to 'bp_interface' i.e. w/o the 's'.
- Added more VMs, to support the larger number of VMs needed for the topo.
- Added the missing leaf template file for the topo.
- Fixed the synthesis of the MACs used in ansible roles so that it does not error out after 256 interfaces.

Additionally, this PR fixed the 'bp_interfaces' for t0-isolated-d128u128s1 yml file.

Summary:
Fixes # (issue)

### Type of change
- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### What is the motivation for this PR?
To achieve a functional t0-isolated-d128u128s2 for use in sonic-mgmt tests.

#### How did you do it?
Made a series of fixes/updates as detailed in the summary above.

#### How did you verify/test it?
Confirmed that the testbed server can now be successfully configured as a result of these changes, and subsequently ran the sonic-mgmt test suite.

### CODE REVIEW

File: ansible/roles/test/files/helpers/change_mac.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ INTF_LIST=$(ls /sys/class/net | grep -E "^eth[0-9]+$")
 for INTF in ${INTF_LIST}; do
     ADDR="$(cat /sys/class/net/${INTF}/address)"
     PREFIX="$(cut -c1-15 <<< ${ADDR})"
-    SUFFIX="$(printf "%02x" ${INTF##eth})"
+    SUFFIX="$(printf "%02x" $((${INTF##eth}%256)))"
```
Comment (sdszhang) [inline ansible/roles/test/files/helpers/change_mac.sh:10]: I think we will have duplicated intf MAC once it's over 256?  
Should we change it to something like the following?
```
    PREFIX="$(cut -c1-13 <<< ${ADDR})"
    INTF_ID=${INTF##eth}
    SUFFIX="$(printf "%x:%02x" $(expr ${INTF_ID} / 256) $(expr ${INTF_ID} % 256))"
```

**Comment by sdszhang** [comment]
@r12f for viz.

**Comment by r12f** [comment]
a few issues is already fixed by this PR: https://github.com/sonic-net/sonic-mgmt/pull/15454/files.

this PR is missing the template, because the topo is generated. it will be better to get the generator fixed too.

**Comment by r12f** [comment]
@ccroy-arista , this PR is merged now: https://github.com/sonic-net/sonic-mgmt/pull/15454.

you might need to merge with the master branch, and I believe certain issues will be gone.

File: ansible/roles/test/files/helpers/change_mac.sh (hunk 1 above)
Comment (ccroy-arista) [reply ansible/roles/test/files/helpers/change_mac.sh:10]: Ah I see, you cut just two additional characters from the prefix, so yes your suggestion should work as-is.

Comment (ccroy-arista) [reply ansible/roles/test/files/helpers/change_mac.sh]: Suggestion implemented in latest commit.

**Review by sdszhang — APPROVED** [review]

File: ansible/vars/topo_t0-isolated-d128u128s2.yml [hunk 2]
```diff
@@ -829,7 +829,7 @@ configuration:
       Ethernet1:
         ipv4: 10.0.0.65/31
         ipv6: fc00::82/126
-    bp_interfaces:
+    bp_interface:
```
Comment (r12f) [inline ansible/vars/topo_t0-isolated-d128u128s2.yml:832]: do you mind to also get d128u128s1 updated?

Comment (ccroy-arista) [reply ansible/vars/topo_t0-isolated-d128u128s2.yml:832]: done

**Review by Janetxxx — APPROVED** [review]

**Comment by ccroy-arista** [comment]
A recent merge from a different PR contained a fix that is also present in this PR, hence the conflict with 20245.

I will open up a separate PR for 202405 branch.

**Comment by ccroy-arista** [comment]
PR for manual cherry-pick to 202405 posted: https://github.com/sonic-net/sonic-mgmt/pull/15786

**Pull request merged**